drivers/hmis/lib/form_data/records/inventory.json

```diff
@@ -1,6 +1,26 @@
 {
   "name": "inventory",
   "item": [
+    {
+      "type": "DATE",
+      "link_id": "project-start",
+      "hidden": true,
+      "initial": [
+        {
+          "value_local_constant": "$projectStartDate"
```

Comment:
This is new conceptually. The way this works is that the initial value for this item will be set to whatever the frontend variable `projectStartDate` is (if defined). This allows us to use some external variables in other logic within the form, such as `bounds` (see below) and `enable_when` (see funder.json in this PR).

This adds some tighter coupling to the frontend code, and I think it should be used minimally. We may not need to use it at all for assessments.

The `$` in `$projectStartDate` is optional and does nothing - I just wanted to indicate clearly that this is a frontend variable, not a field path.

Reply:
So this allows the front-end to add a new inventory record with a default start date without asking the backend for a new record?  Is that what it does?
Also, I'm not sure we want new inventory records to default to the project start unless they are the first inventory record.  Often inventory is added, and the start might be years after project operating start.

Reply:
The notes I have say:
> Inventory start and end dates must be within the project operating period

The changes in this file result in the `min` and `max` props being set on the date picker component, such that it's not possible to select Start and End dates that fall outside of the project operating period. It doesn't set default values on the date pickers at all.  (Unless you're editing an existing Inventory record, of course, in which case the Start and End dates are default to whatever the current value is).

Does that answer your question? I'm not sure what you meant by the "asking the backend for a new record" part.

Reply:
yeah, that makes sense, I was just confused.  It's not pre-filling a new InventoryStartDate, just limiting the picker, which is excellent.